### PR TITLE
Load all DLLs in the lib directory for Windows

### LINKED ISSUE #35358: [Discussion] Implement sth like torch_global_deps on Windows
## What is wrong there?
In https://discuss.pytorch.org/t/torch-cat-runtimeerror-error-in-loadlibrarya/71188/5, we can see that it is caused by the fact that `caffe2_nvrtc.dll` is not loaded during initialization. Currently, it will only happen when one trys to use CUDA in the official distribution of Python 3.8. It doesn't apply the changes to `PATH` for DLL search purposes, which makes  `caffe2_nvrtc.dll`  not able to found during runtime.

## Possible solutions
1. Load `caffe2_nvrtc.dll` when possible during initialization.  This is a quick fix, which is not so general.
2. Write the dependencies to a text file and load them one by one during initialization.
3. Load torch_global_deps during initialization. However, there is currently no easy way to force linking against unused shared libraries unless we use sth like `/INCLUDE:[symbol_name]`. Related PR: https://github.com/pytorch/pytorch/pull/34288.
4. Load all DLLs in the lib directory (Looks like a decent solution)

cc @peterjc123 @ezyang @malfet @EscapeZero @kostmo @soumith

### PR DESCRIPTION
**Pull request by peterjc123**
Fixes #35358.

### CODE REVIEW

**Review by malfet — APPROVED** [review]

File: torch/__init__.py [hunk 1]
```diff
@@ -74,7 +74,7 @@
     import glob
     dlls = glob.glob(os.path.join(th_dll_path, '*.dll'))
     for dll in dlls:
-        ctypes.CDLL(lib_path)
+        ctypes.CDLL(dll)
```
Comment (malfet) [inline torch/__init__.py]: Do you know when CDLL unloads the library? When GC collects a variable?
```suggestion
        ctypes.LoadLibrary(dll)
```

Comment (peterjc123) [reply torch/__init__.py]: The library will not be unloaded automatically. We need to manually do that by calling the win32 api FreeLibrary.

Comment (peterjc123) [reply torch/__init__.py]: About the suggested code change, I checked the source code of `ctypes` and they are just the same, except that CDLL specifies the mode to be `nt._LOAD_LIBRARY_SEARCH_DEFAULT_DIRS | nt._LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR`.

**Pull request closed without merging**